lively.ide/text/rich-text-editor-plugin.js

```diff
@@ -34,6 +35,32 @@ export class RichTextPlugin extends EditorPlugin {
     }
   }
 
+  showIconButton () {
+    const iconButton = part(PropertyLabel, {
+      tooltip: 'Insert Icon',
+      fontSize: 14,
+      textAndAttributes: Icon.textAttribute('heart-music-camera-bolt')
+    });
+    const iconButtonHolder = new Morph({
+      fill: Color.rgb(30, 30, 30).withA(0.95),
+      borderRadius: 3,
+      layout: new TilingLayout({
+        hugContentsVertically: true,
+        hugContentsHorizontally: true
+      }),
+      submorphs: [
+        iconButton]
+    });
+    iconButton.master = { auto: PropertyLabel, hover: PropertyLabelHovered };
+    this.textMorph.iconButton = iconButton;
+    this.textMorph.iconButton.onMouseDown = () => this.textMorph.execCommand('add icon at cursor position');
```

Comment:
Can we use a `connect()` binding on `onMouseDown` instead of monkey patching it here? The current solution will for instance block the default behaviour of `onMouseDown` which is, among other things, the application of master components etc.

Reply:
I will change that, but this got me thinking - why is the application of master components intertwined with the `onMouseDown` invocation of morphs? I.e., should we maybe think about dealing with this on an event level, so probably inside of the dispatcher, where then `onMouseDown` is invoked as well?
Not urgent in any case, and maybe I am also not seeing something important here, but relying on the correct invocation of onMouseDown seems like a footgun? I see that this kind of monkeypatching is not super great in any case, but overwriting onMouseDown and forgetting the super call will thus also brick the mater styling? :thinking: